src/Iris/Interactive/Question.hs

```diff
@@ -107,12 +108,12 @@ yesno question defaultAnswer = do
   where
     loop :: IO YesNo
     loop = do
-        Text.putStr $ question <> " (yes/no) "
+        IO.out $ question <> " (yes/no) "
         hFlush stdout
         input <- Text.getLine
         case parseYesNo input of
             Just answer -> pure answer
             Nothing -> do
-                Text.putStrLn $ "I don't understand your answer: '" <> input <> "'"
-                Text.putStrLn "Please, answer yes or no (or y, or n)"
+                IO.out $ "I don't understand your answer: '" <> input <> "'"
+                IO.out "Please, answer yes or no (or y, or n)"
```

Comment:
This should be `outLn` and not just `out`.

Ideally, if we could fully test this function, the tests could've caught this error. But I don't see a good way to test it. I was thinking about mocking `stdin` somehow to provide input. But this looks impossible (or maybe I just don't know how to do this). Tricks like `silently` allow to capture only `stdout` and `stderr` but they don't allow to substitute `stdin` which is a pity.

A proper test for this function would be to write a separate executable and use some scripting-scenario framework where you can specify input and expected output. But this is a lot of work for this small test, but maybe we'll add it later eventually anyway when things stabilise.

> P.S. I remember already writing this comment but I don't see it in the review. Maybe I forgot to submit it 🤔

Reply:
Have you considered a future improvement which would be "take interactive input from file instead of stdin"? (Useful for batch installation scripts etc)

If we were to support such an improvement we'd need to store the input handle in the environment. And that would also allow for unit testing this part easier. 

Thoughts?

Reply:
I wrote multiple CLI tools but I never had the need to test in batch mode from a file 🤔 

You don't need to mock `stdin` in such cases because you can always pipe the content of the file into your program:

```haskell
cat my-commands | my-program
```

For full E2E testing of CLI tools, one can use an external solution like this one:

* https://github.com/simonmichael/shelltestrunner

I can see how storing `stdin` in the environment can make testing easier but it can also make the architecture of a CLI app more complicated, confusing and potentially introduce errors when you read not from the environment `stdin` but from the global one

So I propose just changing `IO.out` to `IO.outLn` here and invest if full integration testing later.